### PR TITLE
[CBRD-23683] Create BrokerHealthCheck thread on demand

### PR DESCRIPTION
**Pull request by quaff**
http://jira.cubrid.org/browse/CBRD-23683

Currently an unnamed broker health check thread will started if CUBRIDDriver class present even not actually connect to any cubrid database, We should defer creation of that thread, and give it a meaningful name.

### CODE REVIEW

File: src/jdbc/cubrid/jdbc/driver/CUBRIDDriver.java [hunk 1]
```diff
@@ -219,6 +217,15 @@ public Connection connect(String url, Properties info) throws SQLException {
 
 	    u_con.setConnectionProperties(connProperties);
 	    u_con.tryConnect();
+
+		if (!BROKER_HEALTH_CHECK_STARTED.get() && BROKER_HEALTH_CHECK_STARTED.compareAndSet(false, true)) {
+			Thread brokerHealthCheck = new Thread(new BrokerHealthCheck());
+			brokerHealthCheck.setName("Broker Health Check");
+			brokerHealthCheck.setDaemon(true);
+			brokerHealthCheck.setContextClassLoader(null);
+			brokerHealthCheck.start();
```
[the remark this answers is not in the document]
Comment (quaff) [reply src/jdbc/cubrid/jdbc/driver/CUBRIDDriver.java:226]: Why not use ATOMIC? AtomicBoolean::compareAndSet will be called only once, and AtomicBoolean::get will be called every time but It's very very fast.
Currently Thread(Broker Health Check) is a daemon thread and cannot be terminated. mssql driver supports terminate background threads if all connection closed: https://github.com/microsoft/mssql-jdbc/issues/1159 , I hope it's helpful to cubrid guys.

[the remark this answers is not in the document]
Comment (airnet73) [reply src/jdbc/cubrid/jdbc/driver/CUBRIDDriver.java:226]: Currently, it is a daemon thread, but I want to change to a background thread and terminate it as a background thread when all users are connection closed.

**Comment by quaff** [comment]
@kisoo-han ping.

**Comment by kisoo-han** [comment]
@quaff 
Thank you for pinging me and your interest on CUBRID.

You proposed start of health check thread at first connection to the DBMS.
For us, jdbc is very sensitive and delegate interface, because most of our
customer accesses CUBRID database thru jdbc connector.

For this reason, it is difficult to accept your proposal as it is.
But if you propose another half, **stop health check thread** when
no more connection is available, then we will seriously consider to accept your proposal.

I think the basic concept of your suggestion is good.

**Pull request closed without merging**